gms/gossiper.cc

```diff
@@ -432,6 +432,24 @@ future<> gossiper::handle_shutdown_msg(inet_address from) {
     });
 }
 
+future<gossip_get_endpoint_states_response>
+gossiper::handle_get_endpoint_states_msg(gossip_get_endpoint_states_request request) {
+    std::unordered_map<gms::inet_address, gms::endpoint_state> map(endpoint_state_map);
+    const auto& application_states_wanted = request.application_states;
+    for (auto& item : map) {
+        endpoint_state& state = item.second;
+        std::map<application_state, versioned_value>& apps = state.get_application_state_map();
+        for (auto it = apps.begin(); it != apps.end(); ) {
+            if (application_states_wanted.count(it->first) > 0) {
+                it++;
+            } else {
+                it = apps.erase(it);
+            }
+        }
```

Comment:
Why copy everything and then erase? Isn't it simpler to copy just what is wanted?

Reply:
I wanted to avoid copying other members manually in endpoint_state.  But yes, we can copy just want we want. I forgot this one in the version I just sent. I will add it in the next round.

Reply:
I see you are still copying.

Reply:
Yes. I missed it in the version https://github.com/scylladb/scylla/pull/6848#issuecomment-662844040. I will fix it in a new version.